services/bonita-theme/bonita-theme-impl/src/main/java/org/bonitasoft/engine/theme/impl/ThemeServiceStartupHelper.java

```diff
@@ -72,22 +76,37 @@ private File unzipTheme() {
                 try {
                     defaultThemeCssZip.close();
                 } catch (final IOException e) {
-                    throw new BonitaRuntimeException(e);
+                    throw new SBonitaRuntimeException(e);
                 }
             }
         }
         return unzippedCssPortalThemeFolder;
     }
 
-    public void createDefaultThemes(final TenantServiceAccessor tenantServiceAccessor) throws IOException, SThemeCreationException {
-        createDefaultMobileTheme(tenantServiceAccessor);
-        createDefaultPortalTheme(tenantServiceAccessor);
+    /**
+     * Create the default Portal and Mobile themes if they do not already exist else do nothing
+     * @throws IOException
+     * @throws SThemeCreationException
+     * @throws SThemeReadException 
+     */
+    public void createDefaultThemes() throws IOException, SThemeCreationException, SThemeReadException {
+    	// Create default Mobile theme if it does not exist
+    	try {
+ 			themeService.getTheme(SThemeType.MOBILE, true);
+ 		} catch (SThemeNotFoundException e) {
```

Comment:
Can't we check the existence of a theme without catching exceptions, which is less efficient?